crates/vm/levm/src/gas_cost.rs

```diff
@@ -823,67 +823,59 @@ pub fn identity(data_size: usize) -> Result<u64, VMError> {
 }
 
 pub fn modexp(
-    exponent: U256,
-    base_size: u64,
-    exponent_size: u64,
-    modulus_size: u64,
+    exponent: &BigUint,
+    base_size: usize,
+    exponent_size: usize,
+    modulus_size: usize,
 ) -> Result<u64, VMError> {
+    let base_size: u64 = base_size
+        .try_into()
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+    let exponent_size: u64 = exponent_size
+        .try_into()
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+    let modulus_size: u64 = modulus_size
+        .try_into()
+        .map_err(|_| PrecompileError::ParsingInputError)?;
+
     let max_length = base_size.max(modulus_size);
     let words = (max_length
         .checked_add(7)
         .ok_or(OutOfGasError::GasCostOverflow)?)
-        / WORD_SIZE_IN_BYTES_U64;
+    .checked_div(8)
+    .ok_or(InternalError::DivisionError)?;
```

Comment:
Why replace the old `/ WORD_SIZE_IN_BYTES_U64`? Did clippy warn you? Because if not I would leave it as it was. Otherwise it is okay.